app/dashboard/src/modules/payments/components/PlanSelector/PlanSelector.tsx

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file
+ *
+ * Plan selector component.
+ */
+import * as React from 'react'
+
+import * as reactQuery from '@tanstack/react-query'
+
+import { usePaywall } from '#/hooks/billing'
+
+import * as backendProvider from '#/providers/BackendProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import { DIALOG_BACKGROUND } from '#/components/AriaComponents'
+
+import * as backendModule from '#/services/Backend'
+
+import * as components from './components'
+import * as componentForPlan from './getComponentForPlan'
+
+/**
+ * The mutation data for the `onCompleteMutation` mutation.
+ */
+interface CreateCheckoutSessionMutation {
+  readonly plan: backendModule.Plan
+  readonly paymentMethodId: string
+  readonly seats: number
+}
+
+/**
+ * Props for {@link PlanSelector}
+ */
+export interface PlanSelectorProps {
+  readonly showFreePlan?: boolean
+  readonly hasTrial?: boolean
+  readonly userPlan?: backendModule.Plan | undefined
+  readonly plan?: backendModule.Plan | null | undefined
+  readonly onSubscribeSuccess?: (plan: backendModule.Plan, paymentMethodId: string) => void
+  readonly onSubscribeError?: (error: Error) => void
+}
+
+/**
+ * Plan selector component.
+ * Shows the available plans and allows the user to subscribe to one.
+ */
+export function PlanSelector(props: PlanSelectorProps) {
+  const {
+    onSubscribeSuccess,
+    onSubscribeError,
+    plan,
+    userPlan,
+    showFreePlan = true,
+    hasTrial = true,
+  } = props
+
+  const { getText } = textProvider.useText()
+  const backend = backendProvider.useRemoteBackendStrict()
+  const { getPaywallLevel } = usePaywall({ plan: userPlan })
+
+  const onCompleteMutation = reactQuery.useMutation({
+    mutationFn: async (mutationData: CreateCheckoutSessionMutation) => {
+      const { id } = await backend.createCheckoutSession({
+        plan: mutationData.plan,
+        paymentMethodId: mutationData.paymentMethodId,
+        quantity: mutationData.seats,
+      })
+
+      return backend.getCheckoutSession(id).then(data => {
+        if (['trialing', 'active'].includes(data.status)) {
+          return data
+        } else {
+          throw new Error(
+            'Session not complete, please contact the support team or try with another payment method.'
```

Comment:
`Session not complete` -> `Checkout not successful`?
also optionally `contact the support team` -> `contact support`

Reply:
Changed to: `The payment was not successful. Please try again or contact support.`